### PR TITLE
Cleanup of threadpool benchmark

### PR DESCRIPTION
**Pull request by Casper-NS**
- Changed the benchmark to be a single function with parameters that is called multiple times, which emulates the 4 previously used functions.
- Made sure all the benchmarks parse the same number of components to ensure consistency.

### CODE REVIEW

**Review by Laursen79 — DISMISSED** [review]

**Review by Laursen79 — DISMISSED** [review]

File: benches/threadpool_bench.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    id: String,
+    c: &mut Criterion,
+    components: &Vec<String>,
+    query: &String,
+    active_cache: bool,
+) {
+    c.bench_function(&id, |b| {
+        b.to_async(FuturesExecutor).iter(|| async {
+            let backend = ConcreteEcdarBackend::default();
+            let responses = (0..512)
```
Comment (CasperStaahl) [inline benches/threadpool_bench.rs:31]: something like:
```
const NUM_OF_REQUESTS = 512
(...)
let responses = (0..NUM_OF_REQUESTS) (...)
```
Would be more clear

File: benches/threadpool_bench.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+use futures::StreamExt;
+
+mod flamegraph_profiler;
+use flamegraph_profiler::FlamegraphProfiler;
+
+static PATH: &str = "samples/json/EcdarUniversity";
+
+fn send_query_with_components(
+    id: String,
+    c: &mut Criterion,
```
Comment (CasperStaahl) [inline benches/threadpool_bench.rs:23]: Maybe rename c to criterion

**Review by CasperStaahl — CHANGES_REQUESTED** [review]
other than comments lgtm.

File: benches/threadpool_bench.rs (hunk 1 above)
Comment (Casper-NS) [reply benches/threadpool_bench.rs]: Sure, it's added

File: benches/threadpool_bench.rs (hunk 2 above)
Comment (Casper-NS) [reply benches/threadpool_bench.rs:25]: idk, its seems to be the standard to just do it this way for criterion, this is also how all other benchmarks in reveaal are written.

Comment (CasperStaahl) [reply benches/threadpool_bench.rs:25]: coolcool

**Review by CasperStaahl — APPROVED** [review]
lgtm

**Review by S3JER — APPROVED** [review]

**Review by henneboy — APPROVED** [review]
Lgtm

**Pull request merged**